test/e2e/incluster/rbac_test.go

```diff
@@ -0,0 +1,185 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e
+// +build e2e
+
+package incluster
+
+import (
+	"context"
+	"crypto/tls"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	contourv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+	contourv1alpha1 "github.com/projectcontour/contour/apis/projectcontour/v1alpha1"
+	"github.com/projectcontour/contour/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func testProjectcontourResourcesRBAC(namespace string) {
+	Specify("Contour ClusterRole is set up to allow access to projectcontour.io API group resources and resource status", func() {
+		f.Fixtures.Echo.Deploy(namespace, "echo")
+
+		otherNS := "another-" + namespace
+		f.CreateNamespace(otherNS)
+		defer f.DeleteNamespace(otherNS, false)
+		f.Certs.CreateSelfSignedCert(otherNS, "delegated-cert", "delegated-cert", "rbac-test.projectcontour.io")
+
+		// HTTPProxy and TLSCertificateDelegation
+		t := &contourv1.TLSCertificateDelegation{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: otherNS,
+				Name:      "rbac",
+			},
+			Spec: contourv1.TLSCertificateDelegationSpec{
+				Delegations: []contourv1.CertificateDelegation{
+					{
+						SecretName:       "delegated-cert",
+						TargetNamespaces: []string{namespace},
+					},
+				},
+			},
+		}
+		require.NoError(f.T(), f.Client.Create(context.TODO(), t))
+
+		p := &contourv1.HTTPProxy{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "rbac",
+			},
+			Spec: contourv1.HTTPProxySpec{
+				VirtualHost: &contourv1.VirtualHost{
+					Fqdn: "rbac-test.projectcontour.io",
+					TLS: &contourv1.TLS{
+						SecretName: otherNS + "/delegated-cert",
+					},
+				},
+				Routes: []contourv1.Route{
+					{
+						Services: []contourv1.Service{
+							{Name: "invalid-service", Port: 80},
+						},
+					},
+				},
+			},
+		}
+		_, success := f.CreateHTTPProxyAndWaitFor(p, e2e.HTTPProxyInvalid)
+		assert.True(f.T(), success)
+
+		// Update HTTPProxy to valid service.
+		require.NoError(f.T(), f.Client.Get(context.TODO(), client.ObjectKeyFromObject(p), p))
+		p.Spec.Routes[0].Services[0].Name = "echo"
+		require.NoError(f.T(), f.Client.Update(context.TODO(), p))
+		assert.Eventually(f.T(), func() bool {
+			if err := f.Client.Get(context.TODO(), client.ObjectKeyFromObject(p), p); err != nil {
+				return false
+			}
+			return e2e.HTTPProxyValid(p)
+		}, time.Second*5, time.Millisecond*20)
+
+		// Check request to app works.
+		res, ok := f.HTTP.SecureRequestUntil(&e2e.HTTPSRequestOpts{
+			Host: "rbac-test.projectcontour.io",
+			TLSConfigOpts: []func(*tls.Config){
+				e2e.OptSetSNI("rbac-test.projectcontour.io"),
+			},
```

Comment:
no need to explicitly specify this since SNI matches the Host (see https://github.com/projectcontour/contour/blob/main/test/e2e/http.go#L173-L174)

Reply:
ah yes, copy paste!